flake.nix

```
@@ -1,13 +1,209 @@
 {
-  outputs = { self }: {
-    nixosModules.default = self.nixosModules.impermanence;
-    nixosModules.impermanence = import ./nixos.nix;
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    home-manager.url = "github:nix-community/home-manager";
+    home-manager.inputs.nixpkgs.follows = "nixpkgs";
+  };
 
-    homeManagerModules.default = self.homeManagerModules.impermanence;
-    homeManagerModules.impermanence = import ./home-manager.nix;
+  outputs = { self, nixpkgs, home-manager }:
+    let
+      inherit (nixpkgs) lib;
+      forAllSystems = lib.genAttrs lib.systems.flakeExposed;
+    in
+    {
+      lib = import ./lib.nix { inherit lib; };
 
-    # Deprecated
-    nixosModule = self.nixosModules.impermanence;
-    nixosModules.home-manager.impermanence = self.homeManagerModules.impermanence;
-  };
+      nixosModules.default = self.nixosModules.impermanence;
+      nixosModules.impermanence = import ./nixos.nix;
+
+      homeManagerModules.default = self.homeManagerModules.impermanence;
+      homeManagerModules.impermanence = import ./home-manager.nix;
+
+      # Deprecated
+      nixosModule = self.nixosModules.impermanence;
+      nixosModules.home-manager.impermanence = self.homeManagerModules.impermanence;
+
+      devShells = forAllSystems
+        (system:
+          let
+            pkgs = nixpkgs.legacyPackages.${system};
+          in
+          {
+            default =
+              pkgs.mkShell {
+                packages = [
+                  pkgs.nixpkgs-fmt
+                ];
+              };
+          }
+        );
+
+      checks = forAllSystems
+        (system:
+          let
+            pkgs = nixpkgs.legacyPackages.${system};
+            mkTest = { name, configuration }:
+              pkgs.nixosTest {
```

`pkgs.nixosTest` has been mostly replaced with `pkgs.testers.runNixOSTest`, see https://github.com/NixOS/nixpkgs/pull/293891, https://github.com/NixOS/nixpkgs/pull/292759, and https://discourse.nixos.org/t/nixostest-vs-runtest/26419/5.
```suggestion
              pkgs.testers.runNixOSTest {
```